extensions/wikia/Chat2/ChatAjax.class.php

```diff
@@ -262,6 +262,8 @@ static public function giveChatMod() {
 	private static function authenticateServer() {
 		global $wgRequest;
 
+		\Wikia\Security\CSRFDetector::markRequestAsSecure( __METHOD__ );
```

Comment:
I'd rather go with sth like:
```
$result = \Wikia\Security\Utils::matchToken( ChatConfig::getSecretToken(), $wgRequest->getVal( 'token' ) );
if ($result) {
  \Wikia\Security\CSRFDetector::markRequestAsSecure( __METHOD__ );
}
return $result
```
so that is the token does not match we don't mark the request as secure